scripts/quests/bastok/Out_of_the_Depths.lua

```diff
@@ -86,22 +86,32 @@ quest.sections =
                 end,
 
                 [309] = function(player, csid, option, npc)
+                    local currentProg = quest:getVar(player, 'Prog')
```

Comment:
If `currentProg` is only used in the places you've changed, you replace a lot of this with:
```
local currentProg = utils.clamp(quest:getVar(player, 'Prog'), 0, 5)
```

and then using `currentProg + 1` like you already have, but without the additional `currentProg < maxProg`